src/main/java/edu/ucla/library/bucketeer/JobFactory.java

```diff
@@ -114,145 +114,211 @@ public Job createJob(final String aName, final File aCsvFile, final boolean aSub
 
-                    // If it's supposed to have a file, check to see if it does and fail if it doesn't
-                    if (item.hasFile() && WorkflowState.EMPTY.equals(item.getWorkflowState())) {
-                        final Optional<File> file = item.getFile();
+                        // If it's supposed to have a file, check to see if it does and fail if it doesn't
+                        if (item.hasFile() && WorkflowState.EMPTY.equals(item.getWorkflowState())) {
+                            final Optional<File> file = item.getFile();
 
-                        if (file.isEmpty()) {
-                            item.setWorkflowState(WorkflowState.MISSING);
-                            error.addMessage(MessageCodes.BUCKETEER_125, item.getID(), job.getName());
-                        } else if (!file.get().exists()) {
-                            error.addMessage(MessageCodes.BUCKETEER_146, item.getID(), file.get());
+                            if (file.isEmpty()) {
+                                item.setWorkflowState(WorkflowState.MISSING);
+                                error.addMessage(MessageCodes.BUCKETEER_125, item.getID(), job.getName());
+                            } else if (!file.get().exists()) {
+                                error.addMessage(MessageCodes.BUCKETEER_146, item.getID(), file.get());
+                            }
                         }
-                    }
 
-                    // If we've had any exceptions so far, mark the object as failed and store the exception
-                    if (error.countMessages() > 0) {
-                        // Let missing be the primary error if there are other issues
-                        if (!WorkflowState.MISSING.equals(item.getWorkflowState())) {
-                            item.setWorkflowState(WorkflowState.FAILED);
+                        // If we've had any exceptions so far, mark the object as failed and store the exception
+                        if (error.countMessages() > 0) {
+                            // Let missing be the primary error if there are other issues
+                            if (!WorkflowState.MISSING.equals(item.getWorkflowState())) {
+                                item.setWorkflowState(WorkflowState.FAILED);
+                            }
+
+                            // TODO: Store the exception in the item (have to change object to make this happen)
                         }
 
-                        // TODO: Store the exception in the item (have to change object to make this happen)
+                        // Add this item to the job's items list
+                        items.add(item);
+                    } else {
+                        job.setMetadataHeader(columns);
                     }
-
-                    // Add this item to the job's items list
-                    items.add(item);
-                } else {
-                    job.setMetadataHeader(columns);
                 }
             }
```

Comment:
You've logged the errors, but you don't return a ProcessingException with the errors from the `createJob` method so that the `LoadCsvHandler` can return a BAD_REQUEST response. We need to let the user know why their CSV didn't initiate a job with any items.

Reply:
Modified JobFactory to throw exception in event of duplicate headers